agent/consul/state/event.go

```diff
@@ -37,13 +63,47 @@ func (e *EventPublisher) PreparePublish(events []stream.Event) error {
 	return nil
 }
 
-// Commit sends any staged events to the relevant listeners. This is called
+// Commit triggers any staged events to be sent to the relevant listeners. This is called
 // via txn.Defer to delay it from running until the transaction has been finalized.
 func (e *EventPublisher) Commit() {
+	e.stagedLock.Lock()
+	defer e.stagedLock.Unlock()
+
+	e.commitCh <- commitUpdate{
```

Comment:
This was to remove the event publishing work from the hot path in case the publishing/acl update logic ever adds a significant amount of time to commits. We talked about doing this to avoid raft commits potentially taking too long